File: Data Structures & Algorithms/rotate-array/submission-2.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        int[] tp = new int[n];

        for (int i = 0; i < n; i++) {
            tp[(i + k) % n] = nums[i];
        }
        for (int i = 0; i < n; i++) {
            nums[i] = tp[i];
        }
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        win = {}
        l = 0

        for r in range(len(nums)):
            if (r - l) > k:
                win.pop(nums[l])
                l += 1
            if nums[r] in win:
                return True
            win[nums[r]] = r
        return False